Fixing warnings for deprecated SDF usage

pvechersky opened the pull request:
Fixing warnings for deprecated SDF usage in the RotorS odometry plugin. Keeping the code compatible with the old version.

ffurrer commented on rotors_gazebo_plugins/src/gazebo_odometry_plugin.cpp:
```diff
@@ -112,6 +112,64 @@ void GazeboOdometryPlugin::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
     gzthrow("[gazebo_odometry_plugin] Couldn't find specified parent link \"" << parent_frame_id_ << "\".");
   }
+
+#if SDF_MAJOR_VERSION >= 3
```
To keep maintainability I would prefer if we add the `.X()` accessors to the `sdf` vector versions.
In the header, something like this (untested) should do:

```C++
#define Vector3 Vector3Sdf
// Include for Vector3
#undef Vector3

class Vector3 : Vector3Sdf {
 public:
  void X() {return this->x;};
};
```

Not sure if this really works, but in my opinion this would be a lot nicer, as we don't have code duplication.

pvechersky commented:
@ffurrer 

I removed the typedefs and instead added a wrapper class for sdf::Vector3 like you suggested.

I added it to 'common.h' in the plugins such it can be potentially used by other plugins.

However, I changed the name of this new class to SdfVector3 so that it is very clear where it comes from and also because calling '#undef Vector3' after the includes would cause some problems: there is no direct include file for the sdf types and calling '#undef Vector3' would also affect things like gazebo::math::Vector3 etc. since their headers are all in chain.

ffurrer commented:
hmm, but now you don't seem to be actually using `ignition::math::Vector3`, or am I missing something?

pvechersky commented:
@ffurrer 

I moved the typedefs and the wrapper class definition to a separate header file, designed to SDF api wrapper.

ffurrer commented:
@burrimi cyptal?

burrimi commented:
thanks a lot! lgtm
